lib/webhook_json_parser.rb

```diff
@@ -0,0 +1,18 @@
+require 'rack/parser'
+require 'json'
+
+module Sinatra
+  class WebhookJsonParser
+    def call(body)
+      data = JSON.parse(body, quirks_mode: true)
+      branch = (
+        data['ref']                                          ||  # github & gitlab
+        data['refChanges'][0]['refId']            rescue nil ||  # stash
+        data['push']['changes'][0]['new']['name'] rescue nil ||  # bitbucket
+        data['resource']['refUpdates'][0]['name'] rescue nil ||  # TFS/VisualStudio-Git
+        data['repository']['default_branch']                     # github tagged release; no ref.
```

Comment:
Why no `rescue nil` here?

Reply:
That's just what we already had in the old webhook. 

One of those things we are planning on touching up